doc/neps/nep-0036-fair-play.rst

```diff
@@ -0,0 +1,150 @@
+
+During a community call on XX/XX/XX the community decided to draft a
+set of guidelines for companies and external developers to use when
+they modify and redistribute NumPy.
+
+Scope
+-----
+
+This document aims to define a minimal set of rules that, when
+followed, will placate the NumPy developers.  Our hope, however, is
+that developers who think that they need to modify NumPy will consider
+contributing to the project, or using one of the many mechanisms we
+have for allowing our APIs to operate on externally defined objects.
+
+We encourage and exhort developers to first discuss their problem with
+the NumPy developers on the `mailing list <>`__ before resorting to
+patching and releasing their own version of the library.
+
+Fair Play Rules
+---------------
+
+1. Do not reuse the NumPy name for projects not developed by the NumPy
+   community.
+
+   At time of writing, there are only two `numpy`-named packages
+   developed by the community: `numpy` and `numpy-financial`.  We ask
+   that external packages not be named, e.g., `mycompany_numpy`.
+
+   NumPy is a trademark owned by NumFOCUS.
+
+2. Do not republish modified versions of NumPy.
+
+   Modified versions of NumPy make it very difficult for the
+   developers to address bug reports, since we typically do not know
+   which parts of NumPy have been modified.
+
+   If you absolutely have to break this rule (and we implore you not
+   to!), then make it absolutely clear in the `__version__` tag that
+   you have modified NumPy, e.g.::
+
+     >>> print(np.__version__)
+     '1.17.2+mycompany.15`
+
+   We understand that minor patches are often required to make a
+   library work under a certain distribution.  E.g., Debian may patch
+   NumPy so that it searches for optimized BLAS libraries in the
+   correct locations.  But we ask that no substantive changes are
+   made.
+
+3. Do not extend NumPy's API footprint.
+
+   If you absolutely have to break rule two, please do not add
+   additional functions to the namespace.  NumPy's API is already
+   quite large, and we are working hard to reduce it where feasible.
+   Having additional functions exposed in distributed versions is
+   confusing for users and developers alike.
+
+4. *DO* use official mechanism to engage with the API.
+
+   Protocols such as `__array_ufunc__
+   <https://numpy.org/neps/nep-0013-ufunc-overrides.html>`__ and
+   `__array_function__
+   <https://numpy.org/neps/nep-0018-array-function-protocol.html>`__
+   were designed to help external packages interact more easily with
+   NumPy.  E.g., the latter allows objects from foreign libraries to
+   pass through NumPy unharmed.  We actively encourage using any of
+   these "officialy sanctioned" mechanisms for overriding or
+   interacting with NumPy.
+
+Questions & Answers
+-------------------
+
+**Q:** We would like to distribute an optimized version of NumPy that
+utilizes MKL.  You recommend against that, so what are we to do?
```

Comment:
This question is misleading, we don't recommend against that. Using MKL as the BLAS/LAPACK backend is perfectly fine. That's an example of what we'd like to see.

You're thinking about, e.g. using `mklfft` under the hood without there being a mechanism within NumPy to do so. That requires discussion with us first, and we'd come to some conclusion (e.g. please contribute a backend mechanism for that).